code/modules/hippie/pool/pool_controller.dm

```diff
@@ -0,0 +1,363 @@
+	if(stat & (BROKEN|NOPOWER))		// unpowered, no shock
+		return 0
+	if(!prob(prb))
+		return 0
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, src)
+	s.start()
+	if(electrocute_mob(user, get_area(src), src, 0.7))
+		return 1
+	else
+		return 0
+
+/obj/machinery/poolcontroller/proc/poolreagent()
+	for(var/turf/open/pool/water/W in linkedturfs)
+		for(var/mob/living/carbon/human/swimee in W)
+			if(beaker && cur_reagent)
+				beaker.reagents.reaction(swimee, VAPOR, 0.03) //3 percent
+				for(var/datum/reagent/R in beaker.reagents.reagent_list)
+					swimee.reagents.add_reagent(R.id, 0.5) //osmosis
+		for(var/obj/objects in W)
+			if(beaker && cur_reagent)
+				beaker.reagents.reaction(objects, VAPOR, 1)
+			reagenttimer = 4
+
+
+/obj/machinery/poolcontroller/process()
+	updatePool() //Call the mob affecting proc)
+	if(timer > 0)
+		timer--
+		updateUsrDialog()
+	if(reagenttimer > 0)
+		reagenttimer--
+	if(stat & (NOPOWER|BROKEN))
+		return
+	else if(reagenttimer == 0 && !drained)
+		poolreagent()
+
+/obj/machinery/poolcontroller/proc/updatePool()
+	if(!drained)
+		for(var/turf/open/pool/water/W in linkedturfs) //Check for pool-turfs linked to the controller.
+			for(var/mob/M in W) //Check for mobs in the linked pool-turfs.
+				if(isobserver(M))
+					continue
+
+				//End sanity checks, go on
+				switch(temperature) //Apply different effects based on what the temperature is set to.
+					if("Scalding") //Burn the mob.
+						M.bodytemperature = min(500, M.bodytemperature + 35) //heat mob at 35k(elvin) per cycle
+						// if(M.bodytemperature >= 400 && !M.stat)
+							// M << "<span class='danger'>You're boiling alive!</span>"
+
+					if("Frigid") //Freeze the mob.
+						M.bodytemperature = max(80, M.bodytemperature - 35) //cool mob at -35k per cycle, less would not affect the mob enough.
+						// if(M.bodytemperature <= 215 && !M.stat)
+							// M << "<span class='danger'>You're being frozen solid!</span>"
+
+					if("Normal") //Normal temp does nothing, because it's just room temperature water.
+
+					if("Warm") //Gently warm the mob.
+						M.bodytemperature = min(360, M.bodytemperature + 20) //Heats up mobs till the termometer shows up
+
+					if("Cool") //Gently cool the mob.
+						M.bodytemperature = max(250, M.bodytemperature - 20) //Cools mobs till the termometer shows up
+				var/mob/living/carbon/human/drownee = M
+				if(drownee.stat == DEAD)
+					continue
+				if(drownee && drownee.lying && !drownee.internal)
+					if(drownee.stat != CONSCIOUS)
+						drownee.adjustOxyLoss(9)
+						drownee << "<span class='danger'>You're quickly drowning!</span>"
+					else
+						if(!drownee.internal)
+							drownee.adjustOxyLoss(4)
+							if(prob(35))
+								drownee << "<span class='danger'>You're lacking air!</span>"
+
+			for(var/obj/effect/decal/cleanable/decal in W)
+				if(bloody < 800)
+					animate(decal, alpha = 10, time = 20)
+					spawn(25)
```

Comment:
I think there is a proc to qdel something in X amount of time actually, not sure.